tests/basic/pool_function_test.go

```diff
@@ -186,3 +189,153 @@ var _ = Describe("{PoolExpandWithPXRestart}", func() {
+			// Verify error on pool expansion failure
+			var errMatch error
+			re := regexp.MustCompile(fmt.Sprintf(".*failed to find storage pool with UID.*%s.*",
+				invalidPoolUUID))
+			if !re.MatchString(fmt.Sprintf("%v", resizeErr)) {
+				errMatch = fmt.Errorf("failed to verify failure using invalid PoolUUID [%v]", invalidPoolUUID)
+			}
+			dash.VerifyFatal(errMatch, nil, "Pool expand with invalid PoolUUID completed?")
+		})
+	})
+
+})
+
+var _ = Describe("{PoolExpandDiskAddAndVerifyFromOtherNode}", func() {
+
+	BeforeEach(func() {
+		StartTorpedoTest("PoolExpandDiskAddAndVerifyFromOtherNode",
+			"Initiate pool expansion and verify from other node", nil, testrailID)
+		contexts = scheduleApps()
+	})
+
+	JustBeforeEach(func() {
+		poolIDToResize = pickPoolToResize()
+		log.Infof("Picked pool %s to resize", poolIDToResize)
+		poolToBeResized = getStoragePool(poolIDToResize)
+	})
+
+	JustAfterEach(func() {
+		AfterEachTest(contexts)
+	})
+
+	AfterEach(func() {
+		appsValidateAndDestroy(contexts)
+		EndTorpedoTest()
+	})
+
+	stepLog := "should get the existing pool and expand it by adding a disk and verify from other node"
+	log.InfoD(stepLog)
+	It(stepLog, func() {
+		storageNode, err = GetNodeWithGivenPoolID(poolIDToResize)
+		log.FailOnError(err, "Failed to get node with given pool ID")
+
+		// get original total size
+		provisionStatus, err := GetClusterProvisionStatusOnSpecificNode(*storageNode)
+		var orignalTotalSize float64
+		for _, pstatus := range provisionStatus {
+			if pstatus.NodeUUID == storageNode.Id {
+				orignalTotalSize += pstatus.TotalSize
+			}
+		}
+
+		originalSizeInBytes = poolToBeResized.TotalSize
+		targetSizeInBytes = originalSizeInBytes + 100*units.GiB
+		targetSizeGiB = targetSizeInBytes / units.GiB
+
+		log.InfoD("Current Size of the pool %s is %d GiB. Trying to expand to %v GiB with type add-disk",
+			poolIDToResize, poolToBeResized.TotalSize/units.GiB, targetSizeGiB)
+		triggerPoolExpansion(poolIDToResize, targetSizeGiB, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK)
+
+		Step("Ensure pool has been expanded to the expected size", func() {
+			err = waitForOngoingPoolExpansionToComplete(poolIDToResize)
+			dash.VerifyFatal(err, nil, "Pool expansion does not result in error")
+			verifyPoolSizeEqualOrLargerThanExpected(poolIDToResize, targetSizeGiB)
+		})
+
+		stNodes, err := GetStorageNodes()
+		log.FailOnError(err, "Unable to get the storage nodes")
+		var verifyNode node.Node
+		for _, node := range stNodes {
+			status, _ := IsPxRunningOnNode(&node)
+			if node.Id != storageNode.Id && status {
+				verifyNode = node
+			}
+		}
+
+		// get final total size
+		provisionStatus, err = GetClusterProvisionStatusOnSpecificNode(verifyNode)
+		var finalTotalSize float64
+		for _, pstatus := range provisionStatus {
+			if pstatus.NodeUUID == storageNode.Id {
+				finalTotalSize += pstatus.TotalSize
+			}
+		}
+		verifyPoolSizeEqualOrLargerThanExpected(poolIDToResize, uint64(orignalTotalSize))
```

Comment:
This line might not be needed , please delete if not needed

Reply:
done